File: fixtures/scale/dart/large/lib/ciphers/runningkey/runningkey_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'runningkey_cipher.dart';
import 'runningkey_key.dart';
import 'runningkey_vectors.dart';

class RunningKeySuite implements CipherSuite {
  @override
  String name() => 'runningkey';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = RunningKeyCipher(RunningKeyKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in runningkeySamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/runningkey/RunningKeySuite.java
package cryptobench.ciphers.runningkey;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class RunningKeySuite implements CipherSuite {
    @Override
    public String name() {
        return "runningkey";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new RunningKeyCipher(RunningKeyKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : RunningKeyVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}
